test/e2e/specs/default-config/recordedit/domain-filter.spec.js

```diff
@@ -170,15 +170,15 @@ describe("Add a record,", function() {
 
                     chaisePage.recordEditPage.getForeignKeyInputButton(constraintColName, 0).click().then(function () {
                         browser.wait(EC.visibilityOf(modalTitle), browser.params.defaultTimeout);
-
                         return modalTitle.getText();
                     }).then(function(text) {
                         expect(text.indexOf("Choose")).toBeGreaterThan(-1);
 
                         rows = chaisePage.recordsetPage.getRows();
-
+                        browser.sleep(3000);
```

Comment:
This sleep also should not make a difference since you are fetching the rows before this sleep statement. If the problem is that the rows aren't completely proper because they haven't all loaded yet, we should have a `browser.wait()` before fetching the rows.